File: Well.cpp
#include "Well.hpp"

//Constructor, menginisialisasi posisi
Well::Well(pair<int,int> _position) : Facility(_position){

}
/**
 * Interact berguna untuk berinteraksi dengan 'Well'.
 * Berinteraksi dengan 'Well' akan menambah persediaan
 * 'Water' yang dimiliki hingga mencapai maksimal.
 */
void Well::interact(Player* _p){
	_p->setWater(MAX_WATER);
}

File: Java/src/main/java/engisubs/gameobject/cell/facility/Well.java
package engisubs.gameobject.cell.facility;

import java.util.*;

import engisubs.gameobject.Player;

public class Well extends Facility{
	/**
	 * Constructor dari Well
	 * @param  _position Posisi dari Well
	 * @return           Instance dari Well
	 */
    public Well(Map <String, Integer> _position){
      super(_position);
      initGameObject(GameObjectType.WELL, 'W');
    }
    /**
     * Melakukan interact dengan Well untuk mengisi air sampai penuh
     * @param _p Reference dari player untuk melakukan penambahan inventory
     */
    public void interact(Player<?> _p){
      _p.setWater(10);
    }
  }
